src/ios/PurchasesPlugin.swift

```diff
@@ -1,10 +1,397 @@
+        }
+    }
+
+    @objc(invalidatePurchaserInfoCache:) func invalidatePurchaserInfoCache(command: CDVInvokedUrlCommand) {
+        RCCommonFunctionality.invalidatePurchaserInfoCache()
+        self.sendOKFor(command: command)
+    }
+
+    @objc(presentCodeRedemptionSheet:) func presentCodeRedemptionSheet(command: CDVInvokedUrlCommand) {
+        if #available(iOS 14.0, *) {
+            RCCommonFunctionality.presentCodeRedemptionSheet()
+        } else {
+            Logger.warn("[Purchases] Warning: tried to present codeRedemptionSheet, but it's only available on iOS 14.0 or greater.")
+        }
+        self.sendOKFor(command: command)
+    }
+
+    @objc(setAttributes:) func setAttributes(command: CDVInvokedUrlCommand) {
+        guard let attributes = command.arguments[0] as? NSDictionary as? [String: String] else {
+            self.sendBadParameterFor(command: command, parameterNamed: "attributes", expectedType: NSDictionary.self)
+            return
+        }
+
+        RCCommonFunctionality.setAttributes(attributes)
+        self.sendOKFor(command: command)
+    }
+
+    @objc(setEmail:) func setEmail(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "email", setFunction: RCCommonFunctionality.setEmail)
+    }
+
+    @objc(setPhoneNumber:) func setPhoneNumber(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "phoneNumber", setFunction: RCCommonFunctionality.setPhoneNumber)
+    }
+
+    @objc(setDisplayName:) func setDisplayName(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "displayName", setFunction: RCCommonFunctionality.setDisplayName)
+    }
+
+    @objc(setPushToken:) func setPushToken(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "pushToken", setFunction: RCCommonFunctionality.setPushToken)
+    }
+
+    @objc(setProxyURLString:) func setProxyURLString(command: CDVInvokedUrlCommand) {
+        let urlString = command.arguments[0] as? NSString
+        RCCommonFunctionality.proxyURLString = urlString as? String
+        self.sendOKFor(command: command)
+    }
+
+    @objc(setAdjustID:) func setAdjustID(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "adjustID", setFunction: RCCommonFunctionality.setAdjustID)
+    }
+
+    @objc(setAppsflyerID:) func setAppsflyerID(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "appsFlyerID", setFunction: RCCommonFunctionality.setAppsflyerID)
+    }
+
+    @objc(setFBAnonymousID:) func setFBAnonymousID(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "fbAnonymousID", setFunction: RCCommonFunctionality.setFBAnonymousID)
+    }
+
+    @objc(setMparticleID:) func setMparticleID(command: CDVInvokedUrlCommand) {
+        simpleStringSet(command: command, name: "mparticleID", setFunction: RCCommonFunctionality.setMparticleID)
```

Comment:
since it's only used for subs attributes, maybe this could be called `setSubscriberAttribute` or something like that?

Reply:
I like that name better 😄